Src/ILGPU.Algorithms/XMath/Cbrt.cs

```diff
@@ -0,0 +1,229 @@
+﻿// ---------------------------------------------------------------------------------------
+//                                   ILGPU.Algorithms
+//                      Copyright (c) 2021 ILGPU Algorithms Project
+//                                    www.ilgpu.net
+//
+// File: Cbrt.cs
+//
+// This file was made by Marcel Pawelczyk, to be used freely without restriction by
+// the ILGPU project. 
+// This file is part of ILGPU and is distributed under the University of Illinois Open
+// Source License. See LICENSE.txt for details
+// ---------------------------------------------------------------------------------------
+
+namespace ILGPU.Algorithms
+{
+
+    partial class XMath
+    {
+        /// <summary>
+        /// Computes 3rd root of n to within double precision.
+        /// </summary>
+        /// <param name="n">This value's cube root will be calculated.</param>
+        /// <param name="checksafety">Enables safety checks.
+        /// <para>Default : Enabled / true</para>
+        /// <para>This includes checking n for negative and 0 values, Infinities and NaN.</para>
+        /// <para>Set to false at own risk for minor speed improvement to disable safety checks.</para>
+        /// </param>
+        /// <returns>Cbrt(n).</returns>
+        public static unsafe double Cbrt(float n, bool checksafety = true)
+        {
+            // Perform "Safety Check" for; -n, 0, +Inf, -Inf, NaN
+            if (checksafety)
+            {
+                if (n < 0) { return -Cbrt(-n); }
+                if (n == 0 || double.IsNaN(n) || double.IsInfinity(n)) { return n; }
+            }
+
+            // Initial approximation
+            // Convert the binary representation of float into a positive int
+            // Isolate & Convert mantissa into actual power it represents
+            // Perform cube root on 2^P, to appoximate x using power law
+            double x = 1 << (int)((((*(uint*)&n) >> 23) - 127) * 0.33333333333333333333f);
```

Comment:
Do those methods preserve the binary representation of the input? Because that's the key reason I used the unsafe code. Because of the need to convert a float to an int without changing the binary representation.

Also, a thing I can test :)

Reply:
Yes, exactly the intended use case of these methods 👍